src/sources.py

```diff
@@ -167,37 +115,73 @@ def parse_obj_type(obj, c_file, prefix, obj_typename):
         c_file.write('              {\n')
         c_file.write('                return NULL;\n')
         c_file.write('              }\n')
+        if obj.doublearray:
+            c_file.write('              ret->%s_item_lens = calloc(len + 1, sizeof(size_t));\n'
```

Comment:
nit: could you use the existing indentation for the generated C code?

It would be: `calloc (len + 1, sizeof (size_t))`

The same comment applies to the remaining lines

Reply:
of course, i will use existing indentation later. 
how about the way of implementation ? if ok, i will finish left partion, and format the code.

Reply:
the rest of the implementation looks fine.

Once you are done with it, could you open a Draft PR for crun where you update just libocispec to use your new version?  In this way we can test whether crun still works fine.

Reply:
OK